dump/mapping/prepbufr_obs_builder.py

```diff
@@ -74,7 +77,19 @@ def _compute_datetime(self, cycleTimeSinceEpoch, dhr):
 
         return dateTime
 
-    def _add_timestamp(self, container: bufr.DataContainer, reference_time: np.datetime64) -> np.array:
-        cycle_times = np.array([3600 * t for t in container.get('obsTimeMinusCycleTime')]).astype('timedelta64[s]')
-        time = (reference_time + cycle_times).astype('datetime64[s]').astype('int64')
-        container.add('timestamp', time, ['*'])
+    def _replace_timestamp(self, container: bufr.DataContainer, reference_time: np.datetime64) -> np.array:
+        times = container.get('obsTimeMinusCycleTime')
+        time_list = []
+    
+        for t in times:
+            if ma.is_masked(t):
+                timestamp = np.int64(0)
+            else:
+                cycle_time = np.timedelta64(int(3600 * t), 's')
+                timestamp = reference_time + cycle_time
+    
+            time_list.append(timestamp)
+    
+        # Convert to UNIX time (seconds since epoch)
+        time_array = np.array(time_list, dtype='datetime64[s]').astype('int64')
+        container.replace('timestamp', time_array)
```

Comment:
I don't think this is correct as `ma.is_masked` checks a masked array for any masked values (not a single value). You shouldn't need to do this anyways...

Try this:
```python
cycle_times = np.timedelta64(int(3600 * times), 's')
timestamps = ma.masked_array(reference_time + cycle_times, 
                             mask=times.mask, dtype='datetime64[s]').astype('int64')

container.replace('timestamp', timestamp.filled())
```

It might not be neccessary to even spec the mask on timestamp.. Is cycle_times masked?

Note, I should update container.replace to automatically get filled data.